frontend/packages/kubevirt-plugin/src/actions/provider.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { CommonActionFactory } from '@console/app/src/actions/creators/common-factory';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceFor } from '@console/internal/module/k8s';
+import { useK8sModel } from '@console/shared/src/hooks/useK8sModel';
+import { VmActionFactory, VmImportActionFactory } from '../components/vms/menu-actions';
+import { StatusGroup } from '../constants/status-group';
+import { useVMStatus } from '../hooks/use-vm-status';
+import { VirtualMachineImportModel, VirtualMachineInstanceModel } from '../models';
+import { kubevirtReferenceForModel } from '../models/kubevirtReferenceForModel';
+import {
+  isVMCreated,
+  isVMExpectedRunning,
+  isVMRunningOrExpectedRunning,
+} from '../selectors/vm/selectors';
+import { isVMIPaused } from '../selectors/vmi';
+import { VMIKind, VMKind } from '../types';
+
+export const useVmActionsProvider = (vm: VMKind) => {
+  const [k8sModel, inFlight] = useK8sModel(referenceFor(vm));
+  const {
+    metadata: { name, namespace },
+  } = vm;
+  const [vmi] = useK8sWatchResource<VMIKind>({
+    kind: kubevirtReferenceForModel(VirtualMachineInstanceModel),
+    name,
+    namespace,
+    isList: false,
+  });
+  const vmStatusBundle = useVMStatus(name, namespace);
+  const actions = React.useMemo(() => {
+    return vmStatusBundle
```

Comment:
this whole section is not very readable. those spreads can cause performance issues, can you create a dictionary wrapped with memo if needed?
we also need to consider the new `printableStatus` that the VM yaml has.
you can check if the `printableStatus` is available like this.

https://github.com/openshift/console/pull/9761/files#diff-2edf5c3c0f283a0fe7f301625abdeb74011fe199291674e0119379b3100415eeR106

The available statuses as of now are
https://github.com/kubevirt/kubevirt/pull/5528/files#diff-5469726ef6ef072d9258db4593eb8811363099d9181a4e5dbd119af7b2b1414eR1137

Reply:
I don't understand how a dictionary would help here, action provider expects an array of actions.

Also, Do we have a hook that returns these statuses similar to `useVMStatus`, or can I get these from the status of VMKind resource when flag `FLAG_KUBEVIRT_HAS_PRINTABLESTATUS` is true?